test/e2e/tests/settings/change-language.spec.ts

```diff
@@ -63,6 +63,10 @@ describe('Settings - general tab @no-mmi', function (this: Suite) {
 
         await driver.refresh();
 
+        await driver.assertElementNotPresent('.loading-overlay__spinner');
+
+        await driver.findVisibleElement(selectors.labelSpanish);
+
```

Comment:
hi @hjetpoluru my suggestion is to add this conditions before doing the `refresh`. Otherwise we'd still can run into the race condition when refreshing. Let me know if this makes sense :pray:

Reply:
Oops! I thought it was needed after the refresh and before clicking on the dropdown again, cause the overlay appears.

@seaona reading this comment, I feel it would be better to include it both before(just wait for overlay) and after(overlay is gone and the label is visible). Please let me know.

Reply:
await changeLanguage(driver, languageIndex);
       await driver.assertElementNotPresent('.loading-overlay__spinner');

        // Validate the label changes to Spanish
        const isLanguageLabelChanged = await driver.isElementPresent(
          selectors.labelSpanish,
        );
        assert.equal(isLanguageLabelChanged, true, 'Language did not change');

        await driver.refresh();

        await driver.assertElementNotPresent('.loading-overlay__spinner');
        await driver.findVisibleElement(selectors.labelSpanish);

Reply:
yeah I think it's okay. I would keep the labelSpanish before the refresh, since this needs to happen, to ensure we don't refresh before fetching the language change.

We do have a validation that the language [changed below](https://github.com/MetaMask/metamask-extension/pull/27590/files#diff-33bde73bde726ff52c450d10976602e54a3d6c7cabe7637dc7aaab7568e0e08cR83), so I think we can remove the labelSpanish after the refresh (mainly doing the same thing)
  
```

   await changeLanguage(driver, languageIndex);
   await driver.assertElementNotPresent('.loading-overlay__spinner');
   await driver.waitForSelector(selectors.labelSpanish);

    // Validate the label changes to Spanish
    const isLanguageLabelChanged = await driver.isElementPresent(
      selectors.labelSpanish,
    );
    assert.equal(isLanguageLabelChanged, true, 'Language did not change');

    await driver.refresh();

    await driver.assertElementNotPresent('.loading-overlay__spinner');
  ```
  
  let me know if this makes sense!

Reply:
Thanks @seaona for guiding me through this change. That makes sense and I will make the change and push up.